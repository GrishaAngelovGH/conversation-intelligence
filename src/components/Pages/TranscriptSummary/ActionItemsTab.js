import PropTypes from 'prop-types'

export const EmailSuggestion = ({ content }) => (
  <div className="row mt-3 border rounded p-3">
    <div className="col-md-12">
      <i className="bi bi-envelope"></i>
      <span className="fw-bold ms-3">Email Suggestion</span>
      <h5>{content}</h5>
    </div>
  </div>
)

EmailSuggestion.propTypes = {
  content: PropTypes.string.isRequired
}

const ActionItemsTab = () => (
  <div className="row">
    <div className="col-md-12">
      <EmailSuggestion content='Madison Butler to send email about "the quote"' />
      <EmailSuggestion content='Madison Butler to send email about "presentation"' />
    </div>
  </div>
)

export default ActionItemsTab