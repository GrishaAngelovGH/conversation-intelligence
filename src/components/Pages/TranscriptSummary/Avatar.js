const Avatar = ({ content }) => (
  <div className="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold" style={{ width: 40, height: 40 }}>
    {content}
  </div>
)

export default Avatar