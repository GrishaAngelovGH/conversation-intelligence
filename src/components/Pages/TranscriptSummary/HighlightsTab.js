const HighlightsTab = () => (
  <div className="row">
    <div className="col-md-12">
      <h4>Keywords Mentioned</h4>
      <div className="badge bg-primary p-2">quote (2)</div>

      <h4 className="mt-3">People</h4>
      <div className="badge bg-primary p-2">John (1)</div>
      <div className="badge bg-primary p-2 ms-3">Peter (1)</div>

      <h4 className="mt-3">Times</h4>
      <div className="badge bg-primary p-2">yesterday (1)</div>
      <div className="badge bg-primary p-2 ms-3">today (2)</div>
    </div>
  </div>
)

export default HighlightsTab