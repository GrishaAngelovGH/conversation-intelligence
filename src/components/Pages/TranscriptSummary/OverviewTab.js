import Attendee from "../common/Attendee"

const OverviewTab = () => (
  <div className="row">
    <div className="col-md-12">
      <div className="row justify-content-between">
        <div className="col-md-5">
          <span className="fw-bold">Related Records</span>
          <div>
            <i className="bi bi-journal-check"></i>
            <i className="bi bi-bookmark ms-3"></i>
            <i className="bi bi-telephone-plus ms-3"></i>
            <i className="bi bi-inboxes ms-3"></i>
          </div>
        </div>
      </div>

      <div className="row mt-3 justify-content-between">
        <div className="col-md-12">
          <div className="fw-bold">Tags</div>
          <button className="btn btn-primary">Add call tags <i className="bi bi-plus-circle"></i></button>
        </div>
      </div>

      <div className="row mt-3 justify-content-between">
        <div className="col-md-12">
          <div className="fw-bold">Attendees</div>
          <Attendee shortName="JS" fullName="James Smith" />
          <Attendee shortName="IG" fullName="Isabel Garcia" />
          <Attendee shortName="MB" fullName="Madison Butler" />
        </div>
      </div>
    </div>
  </div>
)

export default OverviewTab