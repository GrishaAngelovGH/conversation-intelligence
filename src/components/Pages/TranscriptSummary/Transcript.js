import PropTypes from "prop-types"

import Attendee from "./Attendee"

const TranscriptRow = ({ attendee, duration, content }) => (
  <div className="row border rounded mt-3 p-1 bg-light">
    <div className="col-md-12">
      <div className="row">
        <div className="col-md-12">
          <Attendee shortName={attendee.shortName} fullName={attendee.fullName} />

          <div className="row mt-1 align-items-center">
            <div className="col-md-2">
              {duration}
            </div>
            <div className="col-md-10">
              {content}
            </div>
          </div>
        </div>
      </div>
    </div>
  </div >
)

TranscriptRow.propTypes = {
  attendee: PropTypes.object.isRequired,
  duration: PropTypes.string.isRequired,
  content: PropTypes.element.isRequired
}

const Transcript = () => {

  const transcript = [
    { id: 1, attendee: { shortName: 'JS', fullName: 'James Smith' }, duration: '00:10', content: (<div>Good morning, everyone.</div>) },
    { id: 2, attendee: { shortName: 'JS', fullName: 'James Smith' }, duration: '00:15', content: (<div>Hi, <span className="fw-bold">Madison</span>! How are you?</div>) },
    { id: 3, attendee: { shortName: 'MB', fullName: 'Madison Butler' }, duration: '00:05', content: (<div>Very good, thanks !</div>) },
    { id: 4, attendee: { shortName: 'MB', fullName: 'Madison Butler' }, duration: '00:17', content: (<div>Oh, before I forget, can you please send me the <span className="fw-bold">quote</span> we discussed <span className="fw-bold">yesterday</span>?</div>) },
    { id: 5, attendee: { shortName: 'JS', fullName: 'James Smith' }, duration: '00:04', content: (<div>Sure, I will send you an email with the <span className="fw-bold">quote</span></div>) }
  ]

  return (
    <div className="row p-3">
      <div className="col-md-12">
        <h3 className="text-center">Transcript</h3>
        {
          transcript.map(v => (
            <TranscriptRow key={v.id} attendee={v.attendee} duration={v.duration} content={v.content} />
          ))
        }
      </div>
    </div>
  )
}

export default Transcript