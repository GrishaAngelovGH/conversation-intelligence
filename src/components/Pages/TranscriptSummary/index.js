import TranscriptSummary from './TranscriptSummary'

export default TranscriptSummary