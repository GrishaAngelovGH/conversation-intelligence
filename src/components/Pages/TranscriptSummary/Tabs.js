import Tab from "react-bootstrap/Tab"
import Tabs from "react-bootstrap/Tabs"

import OverviewTab from "./OverviewTab"
import ActionItemsTab from "./ActionItemsTab"
import HighlightsTab from "./HighlightsTab"

const TabsWrapper = () => (
  <Tabs defaultActiveKey="overview" id="tabs-id" className="mb-3"   >
    <Tab eventKey="overview" title="Overview">
      <OverviewTab />
    </Tab>
    <Tab eventKey="action-items" title="Action Items">
      <ActionItemsTab />
    </Tab>
    <Tab eventKey="highlights" title="Highlights">
      <HighlightsTab />
    </Tab>
  </Tabs>
)

export default TabsWrapper