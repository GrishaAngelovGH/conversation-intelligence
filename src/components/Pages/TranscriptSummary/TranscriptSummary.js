import Tabs from "./Tabs"
import Transcript from "./Transcript"

const TranscriptSummary = () => (
  <div className="row g-0 mt-3 justify-content-center">
    <div className="col-md-5">
      <Tabs />
    </div>
    <div className="col-md-5">
      <Transcript />
    </div>
  </div>
)

export default TranscriptSummary