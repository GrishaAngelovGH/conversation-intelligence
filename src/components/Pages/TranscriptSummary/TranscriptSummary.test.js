import { render } from '@testing-library/react'

import TranscriptSummary from './TranscriptSummary'

test('should render component', () => {
  const container = render(<TranscriptSummary />)

  expect(container).toMatchSnapshot()
})