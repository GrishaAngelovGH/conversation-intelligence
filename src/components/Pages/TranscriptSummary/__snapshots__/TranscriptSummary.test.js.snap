// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row g-0 mt-3 justify-content-center"
      >
        <h3
          class="text-center"
        >
          Transcript Summary
        </h3>
        <div
          class="col-md-5"
        >
          <ul
            class="mb-3 nav nav-tabs"
            role="tablist"
          >
            <li
              class="nav-item"
              role="presentation"
            >
              <button
                aria-controls="tabs-id-tabpane-overview"
                aria-selected="true"
                class="nav-link active"
                data-rr-ui-event-key="overview"
                id="tabs-id-tab-overview"
                role="tab"
                type="button"
              >
                Overview
              </button>
            </li>
            <li
              class="nav-item"
              role="presentation"
            >
              <button
                aria-controls="tabs-id-tabpane-action-items"
                aria-selected="false"
                class="nav-link"
                data-rr-ui-event-key="action-items"
                id="tabs-id-tab-action-items"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Action Items
              </button>
            </li>
            <li
              class="nav-item"
              role="presentation"
            >
              <button
                aria-controls="tabs-id-tabpane-highlights"
                aria-selected="false"
                class="nav-link"
                data-rr-ui-event-key="highlights"
                id="tabs-id-tab-highlights"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Highlights
              </button>
            </li>
          </ul>
          <div
            class="tab-content"
          >
            <div
              aria-labelledby="tabs-id-tab-overview"
              class="fade tab-pane active show"
              id="tabs-id-tabpane-overview"
              role="tabpanel"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="row justify-content-between"
                  >
                    <div
                      class="col-md-5"
                    >
                      <span
                        class="fw-bold"
                      >
                        Related Records
                      </span>
                      <div>
                        <i
                          class="bi bi-journal-check"
                        />
                        <i
                          class="bi bi-bookmark ms-3"
                        />
                        <i
                          class="bi bi-telephone-plus ms-3"
                        />
                        <i
                          class="bi bi-inboxes ms-3"
                        />
                      </div>
                    </div>
                    <div
                      class="col-md-5 text-end"
                    >
                      Edit
                    </div>
                  </div>
                  <div
                    class="row mt-3 justify-content-between"
                  >
                    <div
                      class="col-md-12"
                    >
                      <div
                        class="fw-bold"
                      >
                        Tags
                      </div>
                      <button
                        class="btn btn-primary"
                      >
                        Add call tags 
                        <i
                          class="bi bi-plus-circle"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="row mt-3 justify-content-between"
                  >
                    <div
                      class="col-md-12"
                    >
                      <div
                        class="fw-bold"
                      >
                        Attendees
                      </div>
                      <div
                        class="row mt-1 align-items-center"
                      >
                        <div
                          class="col-md-2"
                        >
                          <div
                            class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                            style="width: 40px; height: 40px;"
                          >
                            JS
                          </div>
                        </div>
                        <div
                          class="col-md-5"
                        >
                          <h4
                            class="m-0"
                          >
                            James Smith
                          </h4>
                        </div>
                      </div>
                      <div
                        class="row mt-1 align-items-center"
                      >
                        <div
                          class="col-md-2"
                        >
                          <div
                            class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                            style="width: 40px; height: 40px;"
                          >
                            IG
                          </div>
                        </div>
                        <div
                          class="col-md-5"
                        >
                          <h4
                            class="m-0"
                          >
                            Isabel Garcia
                          </h4>
                        </div>
                      </div>
                      <div
                        class="row mt-1 align-items-center"
                      >
                        <div
                          class="col-md-2"
                        >
                          <div
                            class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                            style="width: 40px; height: 40px;"
                          >
                            MB
                          </div>
                        </div>
                        <div
                          class="col-md-5"
                        >
                          <h4
                            class="m-0"
                          >
                            Madison Butler
                          </h4>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-labelledby="tabs-id-tab-action-items"
              class="fade tab-pane"
              id="tabs-id-tabpane-action-items"
              role="tabpanel"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="row mt-3 border rounded p-3"
                  >
                    <div
                      class="col-md-12"
                    >
                      <i
                        class="bi bi-envelope"
                      />
                      <span
                        class="fw-bold ms-3"
                      >
                        Email Suggestion
                      </span>
                      <h5>
                        Madison Butler to send email about "the quote"
                      </h5>
                    </div>
                  </div>
                  <div
                    class="row mt-3 border rounded p-3"
                  >
                    <div
                      class="col-md-12"
                    >
                      <i
                        class="bi bi-envelope"
                      />
                      <span
                        class="fw-bold ms-3"
                      >
                        Email Suggestion
                      </span>
                      <h5>
                        Madison Butler to send email about "presentation"
                      </h5>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-labelledby="tabs-id-tab-highlights"
              class="fade tab-pane"
              id="tabs-id-tabpane-highlights"
              role="tabpanel"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <h4>
                    Keywords Mentioned
                  </h4>
                  <div
                    class="badge bg-primary p-2"
                  >
                    quote (2)
                  </div>
                  <h4
                    class="mt-3"
                  >
                    People
                  </h4>
                  <div
                    class="badge bg-primary p-2"
                  >
                    John (1)
                  </div>
                  <div
                    class="badge bg-primary p-2 ms-3"
                  >
                    Peter (1)
                  </div>
                  <h4
                    class="mt-3"
                  >
                    Times
                  </h4>
                  <div
                    class="badge bg-primary p-2"
                  >
                    yesterday (1)
                  </div>
                  <div
                    class="badge bg-primary p-2 ms-3"
                  >
                    today (2)
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-md-5"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row g-0 mt-3 justify-content-center"
    >
      <h3
        class="text-center"
      >
        Transcript Summary
      </h3>
      <div
        class="col-md-5"
      >
        <ul
          class="mb-3 nav nav-tabs"
          role="tablist"
        >
          <li
            class="nav-item"
            role="presentation"
          >
            <button
              aria-controls="tabs-id-tabpane-overview"
              aria-selected="true"
              class="nav-link active"
              data-rr-ui-event-key="overview"
              id="tabs-id-tab-overview"
              role="tab"
              type="button"
            >
              Overview
            </button>
          </li>
          <li
            class="nav-item"
            role="presentation"
          >
            <button
              aria-controls="tabs-id-tabpane-action-items"
              aria-selected="false"
              class="nav-link"
              data-rr-ui-event-key="action-items"
              id="tabs-id-tab-action-items"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Action Items
            </button>
          </li>
          <li
            class="nav-item"
            role="presentation"
          >
            <button
              aria-controls="tabs-id-tabpane-highlights"
              aria-selected="false"
              class="nav-link"
              data-rr-ui-event-key="highlights"
              id="tabs-id-tab-highlights"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Highlights
            </button>
          </li>
        </ul>
        <div
          class="tab-content"
        >
          <div
            aria-labelledby="tabs-id-tab-overview"
            class="fade tab-pane active show"
            id="tabs-id-tabpane-overview"
            role="tabpanel"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="row justify-content-between"
                >
                  <div
                    class="col-md-5"
                  >
                    <span
                      class="fw-bold"
                    >
                      Related Records
                    </span>
                    <div>
                      <i
                        class="bi bi-journal-check"
                      />
                      <i
                        class="bi bi-bookmark ms-3"
                      />
                      <i
                        class="bi bi-telephone-plus ms-3"
                      />
                      <i
                        class="bi bi-inboxes ms-3"
                      />
                    </div>
                  </div>
                  <div
                    class="col-md-5 text-end"
                  >
                    Edit
                  </div>
                </div>
                <div
                  class="row mt-3 justify-content-between"
                >
                  <div
                    class="col-md-12"
                  >
                    <div
                      class="fw-bold"
                    >
                      Tags
                    </div>
                    <button
                      class="btn btn-primary"
                    >
                      Add call tags 
                      <i
                        class="bi bi-plus-circle"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="row mt-3 justify-content-between"
                >
                  <div
                    class="col-md-12"
                  >
                    <div
                      class="fw-bold"
                    >
                      Attendees
                    </div>
                    <div
                      class="row mt-1 align-items-center"
                    >
                      <div
                        class="col-md-2"
                      >
                        <div
                          class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                          style="width: 40px; height: 40px;"
                        >
                          JS
                        </div>
                      </div>
                      <div
                        class="col-md-5"
                      >
                        <h4
                          class="m-0"
                        >
                          James Smith
                        </h4>
                      </div>
                    </div>
                    <div
                      class="row mt-1 align-items-center"
                    >
                      <div
                        class="col-md-2"
                      >
                        <div
                          class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                          style="width: 40px; height: 40px;"
                        >
                          IG
                        </div>
                      </div>
                      <div
                        class="col-md-5"
                      >
                        <h4
                          class="m-0"
                        >
                          Isabel Garcia
                        </h4>
                      </div>
                    </div>
                    <div
                      class="row mt-1 align-items-center"
                    >
                      <div
                        class="col-md-2"
                      >
                        <div
                          class="rounded-circle bg-primary d-flex justify-content-center align-items-center text-white fw-bold"
                          style="width: 40px; height: 40px;"
                        >
                          MB
                        </div>
                      </div>
                      <div
                        class="col-md-5"
                      >
                        <h4
                          class="m-0"
                        >
                          Madison Butler
                        </h4>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="tabs-id-tab-action-items"
            class="fade tab-pane"
            id="tabs-id-tabpane-action-items"
            role="tabpanel"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <div
                  class="row mt-3 border rounded p-3"
                >
                  <div
                    class="col-md-12"
                  >
                    <i
                      class="bi bi-envelope"
                    />
                    <span
                      class="fw-bold ms-3"
                    >
                      Email Suggestion
                    </span>
                    <h5>
                      Madison Butler to send email about "the quote"
                    </h5>
                  </div>
                </div>
                <div
                  class="row mt-3 border rounded p-3"
                >
                  <div
                    class="col-md-12"
                  >
                    <i
                      class="bi bi-envelope"
                    />
                    <span
                      class="fw-bold ms-3"
                    >
                      Email Suggestion
                    </span>
                    <h5>
                      Madison Butler to send email about "presentation"
                    </h5>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="tabs-id-tab-highlights"
            class="fade tab-pane"
            id="tabs-id-tabpane-highlights"
            role="tabpanel"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12"
              >
                <h4>
                  Keywords Mentioned
                </h4>
                <div
                  class="badge bg-primary p-2"
                >
                  quote (2)
                </div>
                <h4
                  class="mt-3"
                >
                  People
                </h4>
                <div
                  class="badge bg-primary p-2"
                >
                  John (1)
                </div>
                <div
                  class="badge bg-primary p-2 ms-3"
                >
                  Peter (1)
                </div>
                <h4
                  class="mt-3"
                >
                  Times
                </h4>
                <div
                  class="badge bg-primary p-2"
                >
                  yesterday (1)
                </div>
                <div
                  class="badge bg-primary p-2 ms-3"
                >
                  today (2)
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-5"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
